joomla/joomla-cms
# Small performance check for downgrading utf8mb4

Pull Request for Issue # .

### Summary of Changes
Checks if the string utf8mb4 exists in the query - no need to do a regex replacement otherwise.
How I came across this: had a client that was attempting to install any extension and received the following errors:

```
JInstaller: :Install: Error SQL SQL=
Extension Install: SQL error processing query: DB function reports no errors.
```

The SQL is empty because the database did not support utf8mb4 AND the PCRE library the PHP was using was outdated and didn't understand `(*SKIP)(*FAIL)`. Thus, `preg_replace()` returned false instead of the original `$query`.

It won't make much of a difference in the long run, but I've seen plenty of times this error on the Joomla! forum and perhaps this will solve it.

### Testing Instructions
Install any extension that creates tables in the database. Make sure it installs correctly and the tables are created. Make sure utf8mb4 tables are downgraded to utf8 if the database does not support the character set (MySQL < 5.5 if I recall correctly).

### Documentation Changes Required
None.
